source/common/quic/quic_client_transport_socket_factory.cc

```diff
@@ -32,9 +32,10 @@ QuicClientTransportSocketConfigFactory::createTransportSocketFactory(
   auto quic_transport = MessageUtil::downcastAndValidate<
       const envoy::extensions::transport_sockets::quic::v3::QuicUpstreamTransport&>(
       config, context.messageValidationVisitor());
-  auto client_config = std::make_unique<Extensions::TransportSockets::Tls::ClientContextConfigImpl>(
+  auto client_config_or_error = Extensions::TransportSockets::Tls::ClientContextConfigImpl::create(
```

Comment:
It's best to treat StatusOr like a unique_ptr for the purposes of naming. So:

`absl::StatusOr<Extensions::TransportSockets::Tls::ClientContextConfigImpl> client_config =`

https://abseil.io/tips/181

Reply:
I don't agree with this particular tip and it's not in the style guide so I'd prefer to leave as-is.